base/shell/explorer/trayprop.cpp

```diff
@@ -270,6 +226,107 @@ class CStartMenuSettingsPage : public CPropertyPageImpl<CStartMenuSettingsPage>
     }
 };
 
+class CNotifySettingsPage : public CPropertyPageImpl<CNotifySettingsPage>
+{
+private:
+    HBITMAP m_hbmpTray;
+    HWND m_hwndTaskbar;
+
+    void _UpdateDialog()
+    {
+        BOOL bShowClock = IsDlgButtonChecked(IDC_TASKBARPROP_CLOCK);
+        BOOL bShowSeconds = IsDlgButtonChecked(IDC_TASKBARPROP_SECONDS);
+        BOOL bHideInactive = IsDlgButtonChecked(IDC_TASKBARPROP_HIDEICONS);
+        UINT uImageId;
+
+        HWND hwndCustomizeNotifyButton = GetDlgItem(IDC_TASKBARPROP_ICONCUST);
+        HWND hwndSeconds = GetDlgItem(IDC_TASKBARPROP_SECONDS);
+        HWND hwndTrayBitmap = GetDlgItem(IDC_TASKBARPROP_NOTIFICATIONBITMAP);
+
+        ::EnableWindow(hwndCustomizeNotifyButton, bHideInactive);
+        ::EnableWindow(hwndSeconds, bShowClock);
+        if (!bShowSeconds)
+            CheckDlgButton(IDC_TASKBARPROP_SECONDS, BST_UNCHECKED);
+
+        if (bHideInactive && bShowClock && bShowSeconds)
+            uImageId = IDB_SYSTRAYPROP_HIDE_SECONDS;
+        else if (bHideInactive && bShowClock && !bShowSeconds)
+            uImageId = IDB_SYSTRAYPROP_HIDE_CLOCK;
+        else if (bHideInactive && !bShowClock)
+            uImageId = IDB_SYSTRAYPROP_HIDE_NOCLOCK;
+        else if (!bHideInactive && bShowClock && bShowSeconds)
+            uImageId = IDB_SYSTRAYPROP_SHOW_SECONDS;
+        else if (!bHideInactive && bShowClock && !bShowSeconds)
+            uImageId = IDB_SYSTRAYPROP_SHOW_CLOCK;
+        else if (!bHideInactive && !bShowClock)
+            uImageId = IDB_SYSTRAYPROP_SHOW_NOCLOCK;
+        else
+            ASSERT(FALSE);
```

Comment:
`UNREACHABLE;` since it's impossible that code execution goes there.

Reply:
Agreed, this was a holdover from the original code. I removed it.